app/src/main/java/com/pvp/app/ui/router/Router.kt

```diff
@@ -1,48 +1,123 @@
     routeModifier: Modifier = Modifier,
     routes: List<Route>,
-    scope: CoroutineScope
+    start: Route
 ) {
     NavHost(
         enterTransition = {
-            slideInHorizontally(
-                initialOffsetX = { -it },
-                animationSpec = tween(500)
+            fadeIn(
+                animationSpec = tween(
+                    durationMillis = 300,
+                    easing = LinearEasing
+                )
+            ) + slideIntoContainer(
+                animationSpec = tween(
+                    durationMillis = 300,
+                    easing = EaseOut
+                ),
+                towards = AnimatedContentTransitionScope.SlideDirection.End
             )
         },
         exitTransition = {
-            slideOutHorizontally(
-                targetOffsetX = { it },
-                animationSpec = tween(500)
+            fadeOut(
+                animationSpec = tween(
+                    durationMillis = 300,
+                    easing = LinearEasing
+                )
+            ) + slideOutOfContainer(
+                animationSpec = tween(
+                    durationMillis = 300,
+                    easing = EaseOut
+                ),
+                towards = AnimatedContentTransitionScope.SlideDirection.Start
             )
         },
         modifier = modifier,
         navController = controller,
-        startDestination = destinationStart.path
+        startDestination = when (start) {
+            is Route.Root -> start.start.path
+            is Route.Node -> start.path
+        }
     ) {
-        routes.forEach { r ->
-            composable(route = r.path) {
-                r.screen(
-                    controller,
-                    routeModifier,
-                    scope
-                )
+        routes.forEach { route ->
+            when (route) {
+                is Route.Node -> {
+                    composeRoute(
+                        controller,
+                        route,
+                        routeModifier
+                    )
+                }
+
+                is Route.Root -> {
+                    navigation(
+                        route = route.path,
+                        startDestination = route.start.path
+                    ) {
+                        route.nodes.forEach { node ->
+                            composeRoute(
+                                controller,
+                                node,
+                                routeModifier
+                            )
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+private fun NavGraphBuilder.composeRoute(
+    controller: NavHostController,
+    route: Route.Node,
+    routeModifier: Modifier
+) {
+    composable(route.path) { backstack ->
+        val settled = LocalHorizontalPagerSettled.current
+        var applierRequired by remember(settled) { mutableStateOf(settled) }
```

Comment:
Can't use it within `LaunchedEffect` block, because `LocalRouteOptionsApplier` is a composable function.